### PR TITLE
Add Celestial Coords 1 tutorial notebook [new content]

### LINKED ISSUE #443: Add to WCS tutorial: How to place a scale bar on an image with WCS
A colleague of mine asked me this simple question, and I thought it would make a good addition to the WCS tutorial.  He ended up finding a solution in the source code for AplPy, but I'm certain we could make it simpler. (i.e., we don't need to the if statements to know if our inputs are an Astropy Quantity versus a Unit)

```
## Scale length from my colleague is calculated for a cosmology case
#a2261 is at z = 0.22331. For Omega_M = 0.3, Omega_Lambda = 0.7, this corresponds to 1 arcsec = 3.593 kpc
scale_length = 100.0 / 3.593 * u.arcsec # = 100 kpc

## This code plots that scale length on the image
# The following is taken from AplPy overlays
if isinstance(scale_length, u.Quantity):
    barlength = scale_length.to(u.degree).value
elif isinstance(scale_length, u.Unit):
    barlength = length.to(u.degree)
if wcs.is_celestial:
    pix_scale = wcsutils.proj_plane_pixel_scales(wcs)
else:
    print("you can't do that in this coordinate system")
    pix_scale = [1.0, 1.0]
degrees_per_pixel = np.sqrt(pix_scale[0] * pix_scale[1])
barlength = barlength / degrees_per_pixel
```

### PR DESCRIPTION
**Pull request by kakirastern**
Fixes #443.

Fixes #436 due to a new bug in GitHub. 

Address #400, though not completely fixing it.

Add new FITS World Coordinate System (WCS) tutorial to guide users on how to building one's own `astropy.wcs.WCS` object, serving as an intro to `WCS` using astropy in two ways programmatically. In addition, a simple way to add a scale bar with the `matplotlib` package has been introduced.

### CODE REVIEW

**Comment by kakirastern** [comment]
@eblur @kelle @stargaser Due to some technical issues I needed to open a new PR for the 1st Celestial Coords tutorial. Will follow up here instead moving forward.

**Comment by kakirastern** [comment]
All code pushed to repo. PR is ready for a review.

File: tutorials/notebooks/celestial-coords1/celestial_coords1.ipynb
Comment (eblur) [inline tutorials/notebooks/celestial-coords1/celestial_coords1.ipynb:280]: Please move this to the top of the page with the rest of the imports

File: tutorials/notebooks/celestial-coords1/celestial_coords1.ipynb
Comment (eblur) [inline tutorials/notebooks/celestial-coords1/celestial_coords1.ipynb:458]: Please move this to the top of the page with the rest of the imports

File: tutorials/notebooks/celestial-coords1/celestial_coords1.ipynb
Comment (eblur) [inline tutorials/notebooks/celestial-coords1/celestial_coords1.ipynb]: Add some context to this line:

"First, we will download the FITS file from Zenodo using the Python requests form. Then we will read the FITS file into a variable called header_data_unit."

**Comment by adrn** [comment]
One quick comment: I would suggest renaming this to "Celestial Coordinates with WCS 1: ...". Taking a look at the infrastructure aspects now!

**Comment by kakirastern** [comment]
Thanks @adrn Will make the changes suggested by @eblur now

File: tutorials/notebooks/celestial-coords1/celestial_coords1.ipynb
Comment (kakirastern) [reply tutorials/notebooks/celestial-coords1/celestial_coords1.ipynb]: Sure, done

File: tutorials/notebooks/celestial-coords1/celestial_coords1.ipynb
Comment (kakirastern) [reply tutorials/notebooks/celestial-coords1/celestial_coords1.ipynb]: Sure, done

File: tutorials/notebooks/celestial-coords1/celestial_coords1.ipynb
Comment (kakirastern) [reply tutorials/notebooks/celestial-coords1/celestial_coords1.ipynb]: Sure, done

**Comment by kakirastern** [comment]
> Looks good to me from an infrastructure perspective!
> 
> But: for consistency with other notebooks, could you merge the first cell (the title) with the next cell (header info / authors). You can also remove the "**" (bold), because the header will be bold by default.

Sure, just merged the first two markdown cells, and removed the bolding effect from the headings.

File: tutorials/notebooks/celestial-coords1/celestial_coords1.ipynb
Comment (stargaser) [inline tutorials/notebooks/celestial-coords1/celestial_coords1.ipynb:276]: It would be more straightforward to do:
```
header_data_unit_list = fits.open('https://zenodo.org/record/3946743/files/tailored_dss.22.29.38.50-20.50.13_60arcmin.fits')
```

File: tutorials/notebooks/celestial-coords1/celestial_coords1.ipynb
Comment (stargaser) [inline tutorials/notebooks/celestial-coords1/celestial_coords1.ipynb:279]: This object is an HDUList which has a list-like interface as shown when the header and image are retrieved, so I suggest naming the variable `header_data_unit_list` or maybe `hdulist` with some explanation in the Markdown.

File: tutorials/notebooks/celestial-coords1/celestial_coords1.ipynb
Comment (stargaser) [inline tutorials/notebooks/celestial-coords1/celestial_coords1.ipynb:511]:
```suggestion
    "To add a scale bar to the image of the Helix nebula, we will use the matplotlib `Axes.arrow` method to draw a line. \n",
```

File: pip-requirements.txt
Comment (stargaser) [inline pip-requirements.txt:10]: I think there are a number of packages here that are not needed for this tutorial. Should it be tailored to just those packages that are imported in the notebook? astropy, matplotlib, and requests though I've suggested changes that would remove request.

**Comment by stargaser** [comment]
For the file download part, it would also work to do `from astropy.utils.data import download_file` and then do
```
fname = download_file('https://zenodo.org/record/3946743/files/tailored_dss.22.29.38.50-20.50.13_60arcmin.fits')
hdulist = fits.open(fname)
```
which is more lines of code than opening the URL directly, or using `requests` to write a local file. It would have the advantage of not downloading the file every time the user re-executes the notebook. My own feeling is that using `download_file` is a best practice for Astropy notebooks, but I would like to solicit more opinions on that.

**Comment by kakirastern** [comment]
But I do have reservations about using `astropy.utils.data.download_file` versus `requests` when it comes to Zenodo, as it makes the assumption of FITS files being hosted necessarily. Other than that I have no major objections.

**Comment by eteq** [comment]
Very minor note: I think the `print` functions shouldn't be in there in cases where it's at the end of the cell - instead I'd say just have the output of the cell be the string that's being printed.

File: tutorials/notebooks/celestial-coords1/celestial_coords1.ipynb
Comment (kakirastern) [reply tutorials/notebooks/celestial-coords1/celestial_coords1.ipynb:276]: Sure

File: tutorials/notebooks/celestial-coords1/celestial_coords1.ipynb
Comment (kakirastern) [reply tutorials/notebooks/celestial-coords1/celestial_coords1.ipynb:279]: Okay

File: pip-requirements.txt
Comment (kakirastern) [reply pip-requirements.txt:10]: Oh, I think these are for the whole `astropy-tutorials` package if I am not mistaken. The current way to specify tutorial dependencies are via the `requirements.txt` in the directory for collecting all files for each individual tutorial. So what I have done is to fix a bug in the CI build, hence the changes made.

**Comment by kakirastern** [comment]
> Very minor note: I think the `print` functions shouldn't be in there in cases where it's at the end of the cell - instead I'd say just have the output of the cell be the string that's being printed.

Sure, will make the changes soon.

File: pip-requirements.txt
Comment (stargaser) [reply pip-requirements.txt:10]: Got it, thanks for explaining!

Comment (kakirastern) [reply pip-requirements.txt:10]: You are welcome

**Comment by kakirastern** [comment]
Am changing the temporary downloading path to https://github.com/astropy/astropy-data/raw/6d92878d18e970ce6497b70a9253f65c925978bf/tutorials/celestial-coords1/tailored_dss.22.29.38.50-20.50.13_60arcmin.fits before PR is ready to be merged by maintainers.

File: tutorials/notebooks/celestial-coords1/celestial_coords1.ipynb
Comment (kakirastern) [reply tutorials/notebooks/celestial-coords1/celestial_coords1.ipynb]: Updated with 
```python
header_data_unit_list = fits.open(...)
```

**Comment by kakirastern** [comment]
This is ready for another review before the corresponding https://github.com/astropy/astropy-data/pull/95 PR is merged, since the final link from there will be required for maintainers to merge this PR.

**Review by adrn — APPROVED** [review]
I did my main review via pull request to @kakirastern's branch here, so I'm happy with the notebook now! One minor remaining thing: I think we can remove `requests` from the requirements files?

**Comment by kakirastern** [comment]
> I did my main review via pull request to @kakirastern's branch here, so I'm happy with the notebook now! One minor remaining thing: I think we can remove `requests` from the requirements files?

Yup, done 👍

**Comment by adrn** [comment]
🎉 Thanks @kakirastern !

**Comment by kakirastern** [comment]
Awesome! Appreciate it @adrn 👍

**Pull request merged**
